src/ExtractToFStar.ml

```diff
@@ -898,10 +906,15 @@ and extract_App (ctx : extraction_ctx) (fmt : F.formatter) (inside : bool)
       match qualif.id with
       | Func fun_id ->
           extract_function_call ctx fmt inside fun_id qualif.type_args args
+      | Global global_id ->
+          let fid = A.global_to_fun_id ctx.trans_ctx.fun_context.gid_conv global_id in
+          let fun_id = Regular (A.Regular fid, None) in
+          extract_function_call ctx fmt inside fun_id qualif.type_args args
```

Comment:
Yup, though I'm not sure how I do the extraction, because I want to keep the global bodies & declarations together : but then I can't extract pure functions, then globals. Should I replace `is_global_body` with an optional global id to access it ?

Reply:
No, I think you should simply extract the global bodies upon finding a GlobalDecl, like you do today (you just have to make the global bodies not appear in the list of module declarations, or ignore them, I think)

Reply:
The issue is that it's function declarations that are sorted and visited in order for the translation (with the declaration groups). It worked in the previous version because the globals were the functions, but the order is not preserved for the `global_decl`. I don't know if my explanation was clear ...

Reply:
I now add global declaration groups as well, IMO it's the cleaner way to do it.